caddyconfig/httpcaddyfile/directives.go

```diff
@@ -128,6 +140,59 @@ func RegisterHandlerDirective(dir string, setupFunc UnmarshalHandlerFunc) {
 	})
 }
 
+// RegisterDirectiveOrder registers the default order for a
+// directive from a plugin.
+//
+// This is useful when a plugin has a well-understood place
+// it should run in the middleware pipeline, and it allows
+// users to avoid having to define the order themselves.
+//
+// The directive dir may be placed in the position relative
+// to ('before' or 'after') a directive included in Caddy's
+// standard distribution. It cannot be relative to another
+// plugin's directive.
+func RegisterDirectiveOrder(dir string, position string, standardDir string) {
+	// check if directive was already ordered
+	if directiveIsOrdered(dir) {
+		panic("directive '" + dir + "' already ordered")
+	}
+
+	switch position {
+	case "before":
```

Comment:
Maybe create constants (and a dedicated type?) for these values?

Reply:
Yeah, good point. Wasn't really necessary before because it was only used in the Caddyfile (and Caddyfile config can't refer to Go consts lol)

Reply:
Done :+1: